apps/dashboard/src/components/Invoices/List.tsx

```diff
@@ -1,24 +1,137 @@
-import { Timeline, TimelineItem } from "@litespace/luna";
-import { IInvoice } from "@litespace/types";
-import React, { useMemo } from "react";
-import { Hash } from "react-feather";
-import Invoice from "./Invoice";
+import { IInvoice, IWithdrawMethod, Paginated } from "@litespace/types";
+import React from "react";
+import { Table } from "@/components/common/Table";
+import { ActionsMenu, formatCurrency, useFormatMessage } from "@litespace/luna";
+import { createColumnHelper } from "@tanstack/react-table";
+import { useMemo } from "react";
+import DateField from "@/components/common/DateField";
+import { InfiniteData, UseInfiniteQueryResult } from "@tanstack/react-query";
 
-const List: React.FC<{ invoices: IInvoice.Self[]; onUpdate?: () => void }> = ({
-  invoices,
-  onUpdate,
-}) => {
-  const timeline = useMemo((): TimelineItem[] => {
-    return invoices.map((invoice) => ({
-      id: invoice.id,
-      children: <Invoice invoice={invoice} onUpdate={onUpdate} />,
-      icon: <Hash />,
-    }));
-  }, [invoices, onUpdate]);
+const withdrawMethods = {
+  wallet: "withdraw.methods.wallet",
+  instapay: "withdraw.methods.instapay",
+  bank: "withdraw.methods.bank",
+};
+
+const StatusDescriptions = {
+  pending: "invoices.admin.status.pending",
+  "updated-by-receiver": "invoices.admin.status.updatedByReceiver",
+  "canceled-by-receiver": "invoices.admin.status.canceledByReceiver",
+  "cancellation-approved-by-admin": "invoices.admin.status.canceledByAdmin",
+  fulfilled: "invoices.admin.status.fulfilled",
+  rejected: "invoices.admin.status.rejected",
+};
```

Comment:
- Use the `LocalId` to get some type safety as we might change any of the message ids and we will not get any warning about it and it will cause the dashboard to crash. 
- We use dash naming notation when naming the message id (e.g., `status.updated-by-receiver`)

```suggestion
const statusDescriptionMap: Record<IInvoice.Status, LocalId> = {
  pending: "invoices.admin.status.pending",
  "updated-by-receiver": "invoices.admin.status.updated-by-receiver",
  "canceled-by-receiver": "invoices.admin.status.canceled-by-receiver",
  "cancellation-approved-by-admin": "invoices.admin.status.canceled-by-admin",
  fulfilled: "invoices.admin.status.fulfilled",
  rejected: "invoices.admin.status.rejected",
};
```

Reply:
That's a very nice suggestion 🌊